lib/mapping/CMap.cpp

```diff
@@ -424,7 +424,7 @@ bool CMap::checkForVisitableDir(const int3 & src, const TerrainTile *pom, const
 		if(!vstd::contains(pom->blockingObjects, obj)) //this visitable object is not blocking, ignore
 			continue;
 
-		if (!obj->appearance->isVisitableFrom(src.x - dst.x, src.y - dst.y))
+		if(!obj->appearance || !obj->appearance->isVisitableFrom(src.x - dst.x, src.y - dst.y))

```

Comment:
The question is left if we need this or could we omit calling checkForVisitableDir if there is no template specified?

Reply:
It can be reverted actually, because obj->appearances used in other methods too.
So either we should check for nullptr everywhere or can avoid such changes